lib/config/rules/typescript.js

```diff
@@ -0,0 +1,84 @@
+// See https://github.com/nzakas/eslint-plugin-typescript
+
+module.exports = {
+  // enforces one space after the colon and zero spaces before the colon of a type annotation.
+  'typescript/type-annotation-spacing': ['error'],
+
+  // enforces accessibility modifiers on class properties and methods. (member-access from TSLint)
+  'typescript/explicit-member-accessibility': 'off',
+
+  // enforces interface names are prefixed. (interface-name from TSLint)
+  'typescript/interface-name-prefix': 'off',
+
+  // enforces /// <reference /> is not used. (no-reference from TSLint)
+  'typescript/no-triple-slash-reference': 'error',
+
+  // enforces the any type is not used. (no-any from TSLint)
+  'typescript/no-explicit-any': 'off',
+
+  // enforces the use of as Type assertions instead of <Type> assertions. (no-angle-bracket-type-assertion from TSLint)
+  'typescript/no-angle-bracket-type-assertion': 'error',
+
+  // disallows the use of custom TypeScript modules and namespaces
+  'typescript/no-namespace': 'off',
+
+  // disallows the use of variables before they are defined.
+  'typescript/no-use-before-define': 'off',
+
+  // enforces the use of the keyword namespace over module to declare custom TypeScript modules. (no-internal-module from TSLint)
+  'typescript/prefer-namespace-keyword': 'off',
+
+  // disallows the use of type aliases. (interface-over-type-literal from TSLint)
+  // breaks `export type Message = string | ((colorizer: any) => string);`
+  'typescript/no-type-alias': 'off',
+
+  // enforces a standard member declaration order. (member-ordering from TSLint) <FEEDBACK>
+  'typescript/member-ordering': 'error',
+
+  // prevents TypeScript-specific constructs from being erroneously flagged as unused
+  'typescript/no-unused-vars': 'off',
+
+  // enforces member overloads to be consecutive.
+  'typescript/adjacent-overload-signatures': 'error',
+
+  // disallows parameter properties in class constructors. (no-parameter-properties from TSLint)
+  'typescript/no-parameter-properties': 'off',
+
+  // enforces PascalCased class and interface names. (class-name from TSLint)
+  'typescript/class-name-casing': 'error',
+
+  // enforces a member delimiter style in interfaces and type literals.
+  'typescript/member-delimiter-style': ['error', {
+    delimiter: 'comma',
+    overrides: {
+      typeLiteral: {
+        delimiter: 'comma',
+      },
+    },
+  }],
+
+  // disallows the declaration of empty interfaces. (no-empty-interface from TSLint)
+  'typescript/no-empty-interface': 'off',
+
+
+  // Disable JS specific rules
+  'no-undef': 'off',
+  'import/no-unresolved': 'off',
+  'no-unused-expressions': 'off',
+  'no-unused-vars': 'off',
+  'no-useless-constructor': 'off',
+  'no-shadow': 'off',
+  'no-use-before-define': 'off',
+  'import/no-extraneous-dependencies': 'off',
+
+  // Breaks typescript-eslint-parser
```

Comment:
Can we describe why each of these break it? Have we raised issues for them?

Reply:
Dissecting why each of these breaks with its respective line in our source can be fairly cumbersome. 

Eg. If I re-enable `array-callback-return`, and run the ts config linter in web, the following error shows up:
```
╰─$ yarn run lint
yarn run v1.0.1$ eslint . --ext .ts --ext .tsx --max-warnings 0
Cannot read property 'type' of null
TypeError: Cannot read property 'type' of null
    at EventEmitter.onCodePathStart (/Users/ismailsyed/src/github.com/Shopify/web/node_modules/eslint/lib/rules/array-callback-return.js:193:34)
    at emitTwo (events.js:130:20)
    at EventEmitter.emit (events.js:213:7)
    at processCodePathToEnter (/Users/ismailsyed/src/github.com/Shopify/web/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:360:30)
    at CodePathAnalyzer.enterNode (/Users/ismailsyed/src/github.com/Shopify/web/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:603:9)
    at nodeQueue.forEach.traversalInfo (/Users/ismailsyed/src/github.com/Shopify/web/node_modules/eslint/lib/linter.js:985:32)
    at Array.forEach (<anonymous>)
    at Linter._verifyWithoutProcessors (/Users/ismailsyed/src/github.com/Shopify/web/node_modules/eslint/lib/linter.js:981:19)
    at preprocess.map.textBlock (/Users/ismailsyed/src/github.com/Shopify/web/node_modules/eslint/lib/linter.js:1021:35)
    at Array.map (<anonymous>)
error Command failed with exit code 1.
```

I've been using trial & error to find out which line breaks it. I can do this for every breaking rule if time permits.

Reply:
https://github.com/eslint/typescript-eslint-parser/issues/389